discovery/sync_manager.go

```diff
@@ -0,0 +1,619 @@
+package discovery
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/lightningnetwork/lnd/lnpeer"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing"
+	"github.com/lightningnetwork/lnd/ticker"
+)
+
+const (
+	// DefaultSyncerRotationInterval is the default interval in which we'll
+	// rotate our active syncers for new ones.
+	DefaultSyncerRotationInterval = time.Hour
+
+	// DefaultFullSyncInterval is the default interval in which we'll force
+	// a full sync to ensure we have as much of the public network as
+	// possible.
+	DefaultFullSyncInterval = 24 * time.Hour
```

Comment:
Went with the magic number 6 instead of 24. I've also added a check when `NumActiveSyncers == 0`, which should be the case for light clients, to do a full sync with the first peer they connect to.